compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/main/DefaultIOSMain.uikit.kt

```diff
@@ -16,14 +16,36 @@
 
 package androidx.compose.ui.main
 
-import kotlinx.cinterop.*
-import platform.UIKit.*
-import platform.Foundation.*
+import kotlinx.cinterop.ObjCObjectBase
+import kotlinx.cinterop.autoreleasepool
+import kotlinx.cinterop.cstr
+import kotlinx.cinterop.memScoped
+import kotlinx.cinterop.toCValues
+import platform.Foundation.NSStringFromClass
+import platform.UIKit.UIApplication
+import platform.UIKit.UIApplicationDelegateProtocol
+import platform.UIKit.UIApplicationDelegateProtocolMeta
+import platform.UIKit.UIApplicationMain
+import platform.UIKit.UIResponder
+import platform.UIKit.UIResponderMeta
+import platform.UIKit.UIScreen
+import platform.UIKit.UIViewController
+import platform.UIKit.UIWindow
 
-private var _rootViewController: UIViewController? = null
+private var _makeRootViewController: (() -> UIViewController)? = null
 
-fun defaultUIKitMain(executableName: String, rootViewController: UIViewController) {
-    _rootViewController = rootViewController
+@Deprecated(
+    message = "Replaced by function with postpone controller load",
+    replaceWith = ReplaceWith(
+        expression = "defaultUIKitMain(executableName, makeRootViewController)",
+        imports = arrayOf("androidx.compose.ui.main")
+    )
+)
+fun defaultUIKitMain(executableName: String, rootViewController: UIViewController)
+    = defaultUIKitMain(executableName) { rootViewController }
+
+fun defaultUIKitMain(executableName: String, makeRootViewController: () -> UIViewController) {
```

Comment:
Is it a final look of this API? Name? Package? Parameters?

Also:
- No docs for public API

Reply:
The method marked as deprecated.